src/Knp/Menu/AbstractFactory.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+namespace Knp\Menu;
+
+use Knp\Menu\Factory\CoreExtension;
+use Knp\Menu\Factory\ExtensionInterface;
+
+/**
+ * Class AbstractFactory
+ *
+ * @package Knp\Menu
+ */
+abstract class AbstractFactory implements FactoryInterface
```

Comment:
What is this abstract class ? This looks unrelated to the description of this PR

Reply:
I would like to have a serialized menu object for my app or i would like to write custom extension. In first example for this purpose i'll use JMS\Serializer and I should add annotation to MenuItem. I should override MenuItem and create new CustomMenuFactory, but  i can't use logic in new CustomMenuFactory cause this logic is private

that's ok, Should i  create a new PR?

Reply:
in any case, this should be part of a PR describing your exact needs and explaining what it does, not hidden in a PR talking about refactoring the Twig helper.
and I'd like a clear description of your use case, as I'm not convinced that allowing to replace the item class used by the factory is actually necessary (btw, `JMS\Serializer` supports defining mapping in other formats than annotations, which allows to define mapping for vendor classes)